src/qt/transactiontablemodel.cpp

```diff
@@ -664,7 +664,7 @@ QVariant TransactionTableModel::headerData(int section, Qt::Orientation orientat
 QModelIndex TransactionTableModel::index(int row, int column, const QModelIndex &parent) const
 {
     Q_UNUSED(parent);
-    TransactionRecord* data = priv->index(walletModel->wallet(), walletModel->clientModel().getBestBlockHash(), row);
+    TransactionRecord* data = priv->index(walletModel->wallet(), walletModel->getLastBlockProcessed(), row);
```

Comment:
In commit "qt: Fix regression in TransactionTableModel" (cd3b6d0666fe0e6191596e2aa56980ab60841f2d)

Maybe also add null check on line 195 above:

https://github.com/bitcoin-core/gui/blob/cd3b6d0666fe0e6191596e2aa56980ab60841f2d/src/qt/transactiontablemodel.cpp#L195

```c++
if (!cur_block_hash.IsNull() && rec->statusUpdateNeeded(cur_block_hash) && wallet.tryGetTxStatus(rec->hash, wtx, numBlocks, block_time)) { 
```

To avoid spurious updateStatus calls during shutdown

Reply:
[Updated](https://github.com/bitcoin-core/gui/pull/8#issuecomment-646712144).

Reply:
9735b0750bc1e510a5ca6764f9d3cdca5cc7f760

nit, assert block hash is not null in `TransactionRecord::statusUpdateNeeded`.

Reply:
@promag
> nit, assert block hash is not null in `TransactionRecord::statusUpdateNeeded`.

Does this suggestion implies that `status.needsUpdate` should be the first operand of OR in `TransactionRecord::statusUpdateNeeded()`?

Reply:
No, I don't mean to change that.

Reply:
[Updated](https://github.com/bitcoin-core/gui/pull/8#issuecomment-647760931).